File: src/ClashBoard.Api/Models/Entities/Match.cs
namespace ClashBoard.Api.Models.Entities;

public class Match
{
    public Guid Id { get; set; }
    public Guid LeagueId { get; set; }
    public Guid Player1Id { get; set; }
    public Guid Player2Id { get; set; }
    public Guid? WinnerId { get; set; }
    public MatchStatus Status { get; set; }
    public int Player1Score { get; set; }
    public int Player2Score { get; set; }
    public string? Player1Character { get; set; }
    public string? Player2Character { get; set; }
    public string? Notes { get; set; }
    public DateTime? ScheduledAt { get; set; }
    public DateTime? CompletedAt { get; set; }
    public DateTime CreatedAt { get; set; }

    // Navigation
    public League League { get; set; } = null!;
    public Player Player1 { get; set; } = null!;
    public Player Player2 { get; set; } = null!;
    public Player? Winner { get; set; }
    public ICollection<MatchGame> Games { get; set; } = [];
}

public enum MatchStatus
{
    Scheduled = 0,
    Completed = 1,
    Cancelled = 2
}

File: client/src/types/match.ts
import type { Player } from "./league";

export const MatchStatus = {
  Scheduled: 0,
  Completed: 1,
  Cancelled: 2,
} as const;

export type MatchStatus = (typeof MatchStatus)[keyof typeof MatchStatus];

export interface MatchGame {
  id: string;
  matchId: string;
  gameNumber: number;
  winnerId: string;
  player1Character: string | null;
  player2Character: string | null;
  isPerfect: boolean | null;
}

export interface Match {
  id: string;
  leagueId: string;
  player1Id: string;
  player2Id: string;
  winnerId: string | null;
  status: MatchStatus;
  player1Score: number;
  player2Score: number;
  player1Character: string | null;
  player2Character: string | null;
  notes: string | null;
  scheduledAt: string | null;
  completedAt: string | null;
  createdAt: string;
  player1: Player;
  player2: Player;
  winner: Player | null;
  games: MatchGame[];
}
